spec/controllers/challenge_signups_controller_spec.rb

```diff
@@ -1,27 +1,138 @@
 require 'spec_helper'
 
 describe ChallengeSignupsController, type: :controller do
+  include LoginMacros
+  include RedirectExpectationHelper
+  let(:user) { create(:user) }
+ 
+  let(:closed_challenge) { create(:gift_exchange, :closed) }
+  let(:closed_collection) { create(:collection, challenge: closed_challenge) }
+  let(:closed_signup) { create(:gift_exchange_signup, collection_id: closed_collection.id) }
+  let(:closed_collection_owner) { User.find(closed_collection.all_owners.first.user_id) }
+  let(:closed_signup_owner) { Pseud.find(closed_signup.pseud_id).user }
 
-  let(:tag_set1) { create(:tag_set) }
-  let(:signup) { create(:challenge_signup) }
+  let(:open_challenge) { create(:gift_exchange, :open) }
+  let(:open_collection) { create(:collection, challenge: open_challenge) }
+  let(:open_signup) { create(:gift_exchange_signup, collection_id: open_collection.id) }
+  let(:open_collection_owner) { User.find(open_collection.all_owners.first.user_id) }
+  let(:open_signup_owner) { Pseud.find(open_signup.pseud_id).user }
+
+  describe "new" do
+    it "redirects and errors if sign-up is not open" do
+      fake_login_known_user(user)
+      get :new, params: { collection_id: closed_collection.name, pseud: user.pseuds.first }
+      it_redirects_to_with_error(collection_path(closed_collection),
+                                 "Sign-up is currently closed: please contact a moderator for help.")
+    end
+  end
+
+  describe "show" do
+    xit "redirects and errors if there is no challenge associated with the collection" do
```

Comment:
This test is ignored, does it pass or do we need a follow-up issue for it?

Reply:
The name of the test doesn't match the contents of the test -- the invalid ID below is for the challenge signup, not the challenge itself. (Hence the error message, "What sign-up did you want to work on?")

The sign-up is loaded with `find` instead of `find_by`, which makes it throw an exception instead of returning nil when there's no record: https://github.com/otwcode/otwarchive/blob/540f520aef15d32d57a85e5c9751a2924cea7fa5/app/controllers/challenge_signups_controller.rb#L60-L63

Reply:
Thank you, it was the description that was throwing me off! I've updated that and made an issue.